wallet/addresses.go

```diff
@@ -380,68 +380,45 @@ func (x *xpubAddress) payRevokeCommitment() (script []byte, version uint16) {
 	return s, 0
 }
 
-// addressScript returns an output script paying to address.  This func is
-// always preferred over direct usage of txscript.PayToAddrScript due to the
-// latter failing on unexpected concrete types.
-func addressScript(addr dcrutil.Address) (script []byte, vers uint16, err error) {
-	type scripter interface {
-		PaymentScript() (uint16, []byte)
-	}
-	switch addr := addr.(type) {
-	case scripter:
-		vers, script = addr.PaymentScript()
-	default:
-		script, err = txscript.PayToAddrScript(addr)
-	}
-	return script, vers, err
+// XXX could just remove this.
+func addressScript(addr stdaddr.Address) (script []byte, vers uint16, err error) {
+	vers, script = addr.PaymentScript()
+	return script, vers, nil
 }
 
-func voteRightsScript(addr dcrutil.Address) (script []byte, version uint16, err error) {
+func voteRightsScript(addr stdaddr.Address) (script []byte, version uint16, err error) {
 	switch addr := addr.(type) {
-	case stakeAddress:
-		script, version = addr.voteRights()
+	case stdaddr.StakeAddress:
+		version, script = addr.VotingRightsScript()
 	default:
-		script, err = txscript.PayToSStx(addr)
+		err = errors.E(errors.Invalid, "address cannot be used for voting rights")
 	}
 	return
 }
 
-func ticketChangeScript(addr dcrutil.Address) (script []byte, version uint16, err error) {
+func ticketChangeScript(addr stdaddr.Address) (script []byte, version uint16, err error) {
 	switch addr := addr.(type) {
-	case stakeAddress:
-		script, version = addr.ticketChange()
+	case stdaddr.StakeAddress:
+		version, script = addr.StakeChangeScript()
 	default:
-		script, err = txscript.PayToSStxChange(addr)
+		err = errors.E(errors.Invalid, "address cannot be used for ticket change")
 	}
 	return
 }
 
-func rewardCommitment(addr dcrutil.Address, amount dcrutil.Amount, limits uint16) (script []byte, version uint16, err error) {
+func rewardCommitment(addr stdaddr.Address, amount dcrutil.Amount, limits uint16) (script []byte, version uint16, err error) {
+	// XXX use 1 << 14
```

Comment:
Reminder

Reply:
meant to remove all of these func wrappers. the constant was defined elsewhere and is already being used.